src/main/java/org/terasology/structureTemplates/util/RegionMergeUtil.java

```diff
@@ -161,16 +159,16 @@ public int getMax(BlockRegion r) {
             }
 
             public BlockRegion regionCopyWithMaxSetTo(BlockRegion r, int newMax) {
-                Vector3i max = new Vector3i(newMax, r.getMaxY(), r.getMaxZ());
-                return new BlockRegion().union(r.getMin(new Vector3i())).union(max);
```

Comment:
This snippet is another example for code that does not make use of the potential performance optimizations of re-suing data structures or avoiding creation in the first place.

If we want to keep the fail-safe logic as it is here (with a different semantics than what the method name suggests) I'd rather go for something like this:

```
return new BlockRegion(r.minX(), r.minY(), r.minZ()).union(newMax, r.maxY(), r.maxZ());
```